### ISSUE TITLE
[Vite] The Vite server doesn't restart on `vite.config.js` change

### ISSUE THREAD

**Issue by vursen** [vaadin/flow#12747]
### Description of the bug

In [a basic Vite example](https://stackblitz.com/edit/vitejs-vite-hbzhhv?file=vite.config.ts), you can see that Vite automatically restarts the server once you change the Vite config file that is `vite.config.ts` by default. 

However, for some reason, nothing seems to restart when changing `vite.config.ts` in a Flow / Fusion project.

### Minimal reproducible example

The problem can be seen in any Flow / Fusion project created with https://start.vaadin.com/welcome.

### Expected behavior

The Vite server automatically restarts when `vite.config.ts` changes.

### Actual behavior

The Vite server doesn't restart when `vite.config.ts` changes.

**Comment by xania** [comment]
My workaround for restarting vite on file change:

1. create vite.ts
```javascript
import { createServer } from "vite";
import config from "./vite.config.js";

(async () => {
  const server = await createServer({
    // any valid user config options, plus `mode` and `configFile`
    ...config,
    configFile: false,
  });
  await server.listen();

  server.printUrls();
})();

```

2. start using tsx
>   npx tsx --watch vite.ts

**Comment by Artur-** [comment]
This was fixed in Vite 4.2.0: https://github.com/vitejs/vite/pull/12349
